qiskit_machine_learning/algorithms/classifiers/vqc.py

```diff
@@ -29,27 +33,27 @@
 class VQC(NeuralNetworkClassifier):
     """Variational quantum classifier.
 
-    The variational quantum classifier is a variational algorithm where the
-    measured expectation value is interpreted as the output of a classifier.
+    The variational quantum classifier (VQC) is a variational algorithm where
+    the measured expectation value is interpreted as the output of a
+    classifier.
 
-    Only supports one-hot encoded labels;
-    e.g., data like ``[1, 0, 0]``, ``[0, 1, 0]``, ``[0, 0, 1]``.
+    Constructs a quantum circuit and corresponding neural network, then uses it
+    to instantiate a neural network classifier.
 
-    Multi-label classification is not supported;
-    e.g., data like ``[1, 1, 0]``, ``[0, 1, 1]``, ``[1, 0, 1]``, ``[1, 1, 1]``.
+    Multi-label classification is not supported.
     """
 
     def __init__(
         self,
         num_qubits: int = None,
         feature_map: QuantumCircuit = None,
         ansatz: QuantumCircuit = None,
-        loss: Union[str, Loss] = "cross_entropy",
-        optimizer: Optional[Optimizer] = None,
+        loss: str | Loss = "cross_entropy",
+        optimizer: Optimizer | None = None,
```

Comment:
I am wondering about this. Yes, this seems to be the way forward with its recent introduction, but we now have a mix of code using Union and the vertical bar. This will make things inconsistent over the API Ref. Also I see it changes the formatting for some reason
![image](https://user-images.githubusercontent.com/40241007/163479247-6d59200c-d7f1-4d27-8e3d-b88c0ac29ec1.png)
versus now its 
![image](https://user-images.githubusercontent.com/40241007/163479517-9c1eed03-fdb7-47c4-991b-906187a56242.png)
Plus somehow before it made the first three params Optional - the should be typehinted in the code that way since None is valid as its the default! In the new version that fact has been lost.

I am wondering if it would be "better" to not do this change incrementally and stick to Union for now and perhaps make some overall change later to have consistency through the code/docs rather than some one way some another.

Reply:
Interesting. @manoelmarques could you please also take a look at these weird artifacts in the generated documentation?
I like the new syntax, but if it confuses sphinx, then it is better to keep the old one.

Reply:
I also noted that the current way "fixed" things in regard of some parameters like this should have been declared optional.
> num_qubits: int = None,

where you can see it came out as Optional[int] (I did not know it would do that so was kinda surprised) but it does not do it as int | None in the new way/docs above as you can see - the None aspect is lost. For either way the code typehinting should ideally be fixed of course!

Reply:
@woodsp-ibm So, how do we proceed with the new "|" design in type hints? Keep the old style for definitions and then change them all in one go or update type hints sequentially? I'm favor of updating them in one go.

Reply:
> So, how do we proceed with the new "|" design

Terra has started using this on the primitives - though looking at their API ref docs of them the typehinting does not seem to come out. The docs here seem ok. I wonder if updating everything in one go, though ideal, is practical enough. What we could do is add a prelude section to the release note here about this, i.e say the intent is to migrate to this new style, where previously Optional was used, and some documentation has already been updated in this regard. This would allow things to be a bit more incremental perhaps.

Reply:
Yeah, I notice that now (at least locally) the docs are styled as before, even when coded in the new style (see attached images). Also, given that the changes are already happening incrementally in Optimization and elsewhere, I just left the new-style hints in place.
<img width="809" alt="vqc-parameters-docs" src="https://user-images.githubusercontent.com/14238050/166240829-3d141f9a-1458-4a29-9318-794e89ce0579.png">
<img width="545" alt="vqc-parameters-code" src="https://user-images.githubusercontent.com/14238050/166240838-4b3f2065-94f5-47de-b323-6e772e7f36bd.png">